configure.ac: Add missing comma to atomic_thread_fence check

fweimer-rh opened the pull request:
Without the comma, ntp_cv_func_atomic_thread_fence=yes becomes part of the C source under test, causing compilation and the check to fail.

Found as part of:

* https://fedoraproject.org/wiki/Changes/PortingToModernC
* https://fedoraproject.org/wiki/Toolchain/PortingToModernC

hstenn commented:
Havard, thanks for the patch.  I've attached it to http://bugs.ntp.org/show_bug.cgi?id=3050 .  Is that appropriate?

hstenn commented:
I show this as being fixed in ntp-4.2.8p9.

fweimer-rh committed:
configure.ac: Add missing comma in some checks

Without the comma, (for example) ntp_cv_func_atomic_thread_fence=yes
becomes part of the C source under test, causing compilation and the
check to fail.

Also add a missing int type in the getsockname test.

fweimer-rh commented:
@dlavigne-nwtime Why did you close this PR? As far as I can tell, the bug it fixes is still present in the sources.

dlavigne-nwtime commented:
@fweimer-rh it looks like that PR automatically closed when we deleted the master-no-authorname branch in 2024 as part of branch restructuring to dev and stable and the PR closure got missed. Would you mind reopening the PR with a patch to the stable branch?

hart-NTP commented:
Thanks for the report, Florian.  The missing commas are fixed in configure.ac of 4.2.8p18 circa 2024:

https://github.com/ntp-project/ntp/blob/stable/configure.ac

The getsockname return type addition in sntp/m4/ntp_ipv6.m4 is still pending.  https://bugs.ntp.org/ is the best place to report bugs, but if you resubmit here only that file remains to be fixed.
